packages/back-end/src/jobs/informationSchema.ts

```diff
@@ -0,0 +1,64 @@
+import Agenda, { Job } from "agenda";
+import { DataSourceInterface } from "../../types/datasource";
+import { initializeDatasourceInformationSchema } from "../services/datasource";
+import { logger } from "../util/logger";
+import { updateDataSource } from "../models/DataSourceModel";
+
+const CREATE_INFORMATION_SCHEMA_JOB_NAME = "createInformationSchema";
+type CreateInformationSchemaJob = Job<{
+  datasource: DataSourceInterface;
+  organization: string;
+}>;
+
+let agenda: Agenda;
+export default function (ag: Agenda) {
+  agenda = ag;
+
+  agenda.define(
+    CREATE_INFORMATION_SCHEMA_JOB_NAME,
+    async (job: CreateInformationSchemaJob) => {
+      if (!job.attrs.data) return;
+      const { datasource, organization } = job.attrs.data;
+
+      if (!datasource || !organization) return;
+
+      try {
+        await initializeDatasourceInformationSchema(datasource, organization);
+      } catch (e) {
+        // Update the datasource.settings.informationSchema object to reflect the error.
+        await updateDataSource(datasource.id, organization, {
+          settings: {
+            ...datasource.settings,
+            informationSchema: {
+              id: undefined,
```

Comment:
What if the information schema synced successfully earlier and this is just a refresh that fails?  A stale schema is better than no schema, so we don't want to wipe out any existing data we have.

Reply:
Ooof - good call.